lib/index.ts

```diff
@@ -235,9 +235,10 @@ const removeOmitted = (swaggerFieldSchema: {
 function documentModel(Model, options: { props?: string[], omitId?: boolean } = {}): any {
   let {
     props = [],
+    omitId = true,
   } = options;
   props = [...defaultSupportedMetaProps, ...props];
-  omitted = new Set(options.omitId ? ['__v', '_id'] : ['__v']);
+  omitted = new Set(omitId ? ['__v', '_id'] : ['__v']);
```

Comment:
is there a reason you're creating a new set instead of just adding the keys to the existing set?

Reply:
Yes.
`Omitted` variable is in the outer scope, so if we change it, the new value will be used in next function calls.
By making a new set we drop it back to default every function call.

Reply:
ah good catch -

the omitted variable in the root scope is in use by the function `removeOmitted` also in the root scope.

assigning the new Set to the root level `omitted` variable still globally mutates the root scope.

since `removeOmitted` is only used once within the scope of `documentModel` you could just move that function declaration to be in scope of documentModel and set it once as a fn of defaults + input omitFields

https://github.com/giddyinc/mongoose-to-swagger/blob/3d06ca28d34c09bb1a75b7469826dfcd83ae5aaa/lib/index.ts#L218

https://github.com/giddyinc/mongoose-to-swagger/blob/3d06ca28d34c09bb1a75b7469826dfcd83ae5aaa/lib/index.ts#L228

Reply:
done